src/BloomBrowserUI/bookEdit/bloomField/bloomField.ts

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+    static RequiresParagraphs(field : HTMLElement) : boolean {
+        return $(field).closest('.bloom-requiresParagraphs').length > 0
+                //this signal used to let the css add this conversion after some SIL-LEAD SHRP books were already typed
+            || ($(field).css('border-top-style') === 'dashed');
+    }
+
+    static HandleFieldFocus(field:HTMLElement) {
+
+       BloomField.MoveCursorToFirstParagraph(field, CursorPosition.start);
+        return;
+
+        if (!BloomField.RequiresParagraphs(field)) {
+            return; // nothing to do
+        }
+        if ($(field).text() === '' && $(field).find("p").length === 0) {
+            //stick in a paragraph, which makes FF do paragraphs instead of BRs.
+            $(field).html('<p class="me - initial">&nbsp;</p>');
+            // &zwnj; (zero width non-joiner) would be better but it makes the cursor invisible
+            //now select that space, so we delete it when we start typing
+
+            var el = $(field).find('p')[0].childNodes[0];
+            var range = document.createRange();
+            range.selectNodeContents(el);
+            var sel = window.getSelection();
+            sel.removeAllRanges();
+            sel.addRange(range);
+        } else {
+            var p = $(field).find('p')[0];
+            if (!p) {
+                return; // these have text, but not p's yet. We'll have to wait until they leave (blur) to add in the P's.
+            }
+            var range = document.createRange();
+            range.selectNodeContents(p);
+            range.collapse(true); //move to start of first paragraph
+
+            var sel = window.getSelection();
+            sel.removeAllRanges();
+            sel.addRange(range);
+        }
+    }
+
+
+    static MoveCursorToFirstParagraph(field: HTMLElement, position: CursorPosition ){
+        var range = document.createRange();
+        if(position === CursorPosition.start) {
+            range.selectNodeContents($(field).find('p').first()[0]);
+        }
+        else{
+            range.selectNodeContents($(field).find('p').last()[0]);
+        }
+        range.collapse(position === CursorPosition.start);//true puts it at the start
+        var sel = window.getSelection();
+        sel.removeAllRanges();
+        sel.addRange(range);
+    }
+    static ManageWhatHappensIfTheyDeleteEverything(field: HTMLElement) {
+        // if the user types (ctrl+a, del) then we get an empty element or '<br></br>', and need to get a <p> in there.
+        // if the user types (ctrl+a, 'blah'), then we get
+
+        $(field).on("input", function (e) {
+            if ($(this).find('p').length === 0) {
+                BloomField.ModifyForParagraphMode(this);
+
+                // Now put the cursor in the paragraph, *after* the character they may have just typed or the
+                // text they just pasted.
+                BloomField.MoveCursorToFirstParagraph(field, CursorPosition.end);
```

Comment:
Seems like the field might not be empty (or all selected) before they input something, e.g., if HandleFieldFocus did not switch things to paragraph mode because the field already had content.
And if it's OK to ModifyForParagraphMode in the middle of typing, why isn't it when the field is first focused?